### PR TITLE
feat(webhook): add exponential backoff retry

### PR DESCRIPTION
**Pull request by ArnaudCalmettes**

### CODE REVIEW

File: ee/webhooks/cmd/flag/flags.go [hunk 1]
```diff
@@ -32,21 +36,24 @@ const (
...
+	DefaultRetriesCron = time.Minute
 )
 
 func Init(flagSet *pflag.FlagSet) {
 	flagSet.Bool(Debug, false, "Debug mode")
 	flagSet.String(LogLevel, logrus.InfoLevel.String(), "Log level")
 
 	flagSet.String(Listen, DefaultBindAddressServer, "server HTTP bind address")
-	flagSet.DurationSlice(RetriesSchedule, DefaultRetriesSchedule, "worker retries schedule")
+	flagSet.DurationSlice(RetriesSchedule, nil, "worker retries schedule")
```
Comment (gfyrag) [inline ee/webhooks/cmd/flag/flags.go:47]: You can drop the flag, the service is not tab stable today, and we use env vars to configure services.

File: ee/webhooks/pkg/retry/backoff_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExponentialBackoffNominal(t *testing.T) {
+	policy := NewExponentialBackoff(time.Minute, time.Hour, 24*time.Hour)
```
Comment (gfyrag) [inline ee/webhooks/pkg/retry/backoff_test.go:12]: t.Parallel() :)

File: ee/webhooks/pkg/retry/backoff_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		8 * time.Minute,
+		16 * time.Minute,
+		32 * time.Minute,
+		time.Hour,
+		time.Hour,
+		time.Hour,
+	}
+	for attemptNumber, wantDelay := range wantDurations {
+		t.Run(fmt.Sprint("attempt #", attemptNumber), func(t *testing.T) {
+			gotDelay, err := policy.GetRetryDelay(attemptNumber)
```
Comment (gfyrag) [inline ee/webhooks/pkg/retry/backoff_test.go:26]: t.Parallel() :)

**Review by gfyrag — APPROVED** [review]

File: ee/webhooks/pkg/retry/backoff_test.go (hunk 3 above)
Comment (ArnaudCalmettes) [reply ee/webhooks/pkg/retry/backoff_test.go:27]: I added it.

**Pull request merged**